### PR TITLE
Check for player inventory before looping

### PR DESCRIPTION
**Pull request by nerzhul**
This prevent useless inventory check on each object in the loop

### CODE REVIEW

**Comment by HybridDog** [comment]
I think it's implemented this way because most times there are no collectible items and the check is simply skipped. This is now no longer the case when magnet mode is enabled.

File: init.lua [hunk 1]
```diff
@@ -262,7 +262,12 @@ if legacy_setting_getbool("item_drop.enable_item_pickup",
 
 		local pos = player:get_pos()
 		pos.y = pos.y+0.5
-		local inv
+		local inv = player:get_inventory()
+		if not inv then
+			minetest.log("error", "[item_drop] Couldn't " ..
+				"get inventory")
+			return
+		end
```
Comment (HybridDog) [inline init.lua]: According to lua_api, inv is never nil for players, so this check can be removed or replaced with an assertion.
```suggestion
```

**Comment by nerzhul** [comment]
Thanks for the review, i added your comment. Don't hesitate to amend more & merge

**Comment by HybridDog** [comment]
I've merged it now.

**Pull request merged**